nano/node/lmdb/lmdb.cpp

```diff
@@ -307,6 +310,10 @@ bool nano::mdb_store::do_upgrades (nano::write_transaction & transaction_a, bool
 			upgrade_v20_to_v21 (transaction_a);
 			[[fallthrough]];
 		case 21:
+			upgrade_v21_to_v22 (transaction_a);
```

Comment:
The are no upgrade operations to be performed, but this will be necessary in case we're going to increase the database version number.

Reply:
> The are no upgrade operations to be performed, but this will be necessary in case we're going to increase the database version number.

Using LMDB the reverse_links table must be explicitly created, which is the upgrade operation I implemented in upgrade_v21_to_v22. RocksDB is more dynamic and works just fine.

Reply:
@thsfs Looks like lmdb tables are already created when nano::mdb_store::open_databases is called with MDB_CREATE before upgrades are processed. So there is no extra need for a database upgrade. The last upgrade v20 to v21 just got me confused, because the table for final votes was created there explicitly, but the current design makes this obsolete as well.

Reply:
> @thsfs Looks like lmdb tables are already created when nano::mdb_store::open_databases is called with MDB_CREATE before upgrades are processed. So there is no extra need for a database upgrade. The last upgrade v20 to v21 just got me confused, because the table for final votes was created there explicitly. The current design makes this obsolete as well.

@thsfs @dsiganos Ok, I was wrong. `open_databases (error, transaction, MDB_CREATE);` gets called only if `is_fully_upgraded == false`, which is not the case if we don't increase the version number. Will think about this tomorrow.